src/pip_sync_faster/main.py

```diff
@@ -1,18 +1,75 @@
+import hashlib
```

Comment:
```suggestion
#!/usr/bin/env python
import hashlib
```

Along with the `__main__` section at the bottom this would allow you to call this directly on the command line which might be handy for testing.